runners/google-cloud-dataflow-java/src/test/java/org/apache/beam/runners/dataflow/TestDataflowRunnerTest.java

```diff
@@ -603,8 +584,12 @@ public void testStreamingOnSuccessMatcherWhenPipelineFails() throws Exception {
     when(mockJob.waitUntilFinish(any(Duration.class), any(JobMessagesHandler.class)))
         .thenReturn(State.FAILED);
 
-    runner.run(p, mockRunner);
-    // If the onSuccessMatcher were invoked, it would have crashed here.
+    try {
+      runner.run(p, mockRunner);
+    } catch (RuntimeException e) {
```

Comment:
It actually isn't a perfect fit - this isn't testing that an exception is thrown. Note that if an exception is _not_ thrown this test can still succeed, as long as the `onSuccessMatcher` is not invoked. But that's getting a bit silly about fine-grained testing, and the comment does "expect" the exception. So: done.